src/sdltiles.cpp

```diff
@@ -1053,6 +1058,314 @@ void clear_window_area( const catacurses::window &win_ )
                     win->width * fontwidth, win->height * fontheight, color_as_sdl( catacurses::black ) );
 }
 
+cata::optional<std::pair<tripoint_abs_omt, std::string>> get_mission_arrow(
+            const inclusive_cuboid<tripoint> &overmap_area, const tripoint_abs_omt &center )
+{
+    if( get_avatar().get_active_mission() == nullptr ) {
+        return cata::nullopt;
+    }
+    const tripoint_abs_omt mission_target = get_avatar().get_active_mission_target();
+
+    std::string mission_arrow_variant = "mission_cursor";
+    if( overmap_area.contains( mission_target.raw() ) ) {
+        return std::make_pair( mission_target, mission_arrow_variant );
+    }
+
+    const std::vector<tripoint> mission_trajectory = line_to( center.raw(), mission_target.raw() );
+
+    tripoint prev( INT_MAX, INT_MAX, INT_MAX );
```

Comment:
```suggestion
    cata::optional<tripoint> prev;
```
Instead of using `INT_MAX`